### ISSUE TITLE
JX is using definitions that do not work in newer k8s clusters

### ISSUE THREAD

**Issue by vfarcic** [jenkins-x/jx#7240]
### Summary 

`jx boot` fails with the following message:

```
unable to recognize "/var/folders/73/94ypm_917397bvg1y3f7phkc0000gn/T/helm-template-workdir-711852854/jenkins-x/output/namespaces/jx/env/charts/jenkins-x-platform/charts/chartmuseum/templates/part0-deployment.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "/var/folders/73/94ypm_917397bvg1y3f7phkc0000gn/T/helm-template-workdir-711852854/jenkins-x/output/namespaces/jx/env/charts/jenkins-x-platform/charts/heapster/templates/part0-deployment.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"'
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step helm apply --boot --remote --name jenkins-x --provider-values-dir ../kubeProviders' command in directory 'env', output: ''
```

```bash
cat /var/folders/73/94ypm_917397bvg1y3f7phkc0000gn/T/helm-template-workdir-711852854/jenkins-x/output/namespaces/jx/env/charts/jenkins-x-platform/charts/heapster/templates/part0-deployment.yaml
```

The output:

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: jenkins-x-heapster
  annotations:
    jenkins.io/chart: env
  labels:
    chart: heapster-0.3.2
    jenkins.io/chart-release: jenkins-x
    jenkins.io/namespace: jx
    jenkins.io/version: "1"
spec:
  replicas: 1
  template:
    metadata:
      annotations: {}
      labels:
        app: jenkins-x-heapster
    spec:
      serviceAccountName: jenkins-x-heapster
      containers:
      - name: heapster
        image: k8s.gcr.io/heapster:v1.5.2
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 8082
        livenessProbe:
          httpGet:
            path: /healthz
            port: 8082
          initialDelaySeconds: 180
          timeoutSeconds: 5
        resources:
          limits:
            cpu: 100m
            memory: 128Mi
          requests:
            cpu: 100m
            memory: 128Mi
        command:
        - /heapster
        - --source=kubernetes.summary_api:''
      - name: heapster-nanny
        image: k8s.gcr.io/addon-resizer:1.7
        imagePullPolicy: IfNotPresent
        resources:
          limits:
            cpu: 50m
            memory: 90Mi
          requests:
            cpu: 50m
            memory: 90Mi
        env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: MY_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        command:
        - /pod_nanny
        - --deployment=jenkins-x-heapster
        - --container=heapster
        - --cpu=150m
        - --extra-cpu=10m
        - --memory=200Mi
        - --extra-memory=6Mi
        - --threshold=5
        - --poll-period=300000
```

It's using `Deployment` based on `extensions/v1beta1` which has been deprecated for a long time and now is not even available in newer k8s versions. I'm using `v1.18.1`.

As a side note... Why do we need Heapster? It has been deprecated for years now.

**Comment by ankitm123** [comment]
This is related to the 1.16 upgrade: #6957. https://github.com/jenkins-x/jx/issues/6957#issuecomment-627046222

**Comment by vfarcic** [comment]
Closing this issue since it's already being addressed in #6957
